<a href="https://colab.research.google.com/github/lucygoncalves/lucyborboletaazul/blob/main/Trabalhando_com_Lista_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Trabalhando com Lista em Python

Bem-vindo ao notebook de exemplos práticos, onde trabalhamos com o recurso de listas em **Python!**. As listas são uma das estruturas de dados mais versáteis e amplamente utilizadas em **Python**. Elas permitem armazenar uma coleção ordenada de elementos, que podem ser de diferentes tipos, como números, strings e até mesmo outras listas. Neste notebook, exploraremos diversas operações que podemos realizar com listas, desde a criação e manipulação básica até técnicas avançadas, utilizando os principais métodos disponíveis para este tipo de variável.

Também é possível ir para um tutorial on-line do Python, acesse pelo link [List no Python](https://docs.python.org/3/tutorial/datastructures.html)


## Criando uma Lista e Acessando seus Elementos

In [ ]:
# Criando uma lista
lista = [1, 2, 3, 4, 5]

# Acessando elementos da lista
primeiro_elemento = lista[0]
ultimo_elemento = lista[-1]

print("Lista completa:", lista)
print("Primeiro elemento:", primeiro_elemento)
print("Último elemento:", ultimo_elemento)

## Adicionando e Removendo Elementos

In [ ]:
# Criando uma lista vazia
lista = []

# Adicionando elementos à lista
lista.append(1)
lista.append(2)
lista.append(3)

print("Lista após adicionar elementos:", lista)

Lista após adicionar elementos: [1, 2, 3]


In [ ]:
# Criando uma lista vazia
lista = []

# Adicionando elementos à lista
lista.append(1)
lista.append(2)
lista.append(3)

print("Lista após adicionar elementos:", lista)

# Removendo elementos da lista
elemento_removido = lista.pop()
print("Elemento removido:", elemento_removido)
print("Lista após remover elemento:", lista)

Lista após adicionar elementos: [1, 2, 3]
Elemento removido: 3
Lista após remover elemento: [1, 2]


## Ordenando uma lista

In [ ]:
# Criando uma lista desordenada
lista = [5, 2, 4, 1, 3]
print("Lista sem ordenacao:", lista)
# Ordenando a lista
lista.sort()
print("Lista ordenada crescente:", lista)
lista.sort(reverse=True)
print("Lista ordenada decrescente:", lista)

Lista sem ordenacao: [5, 2, 4, 1, 3]
Lista ordenada crescente: [1, 2, 3, 4, 5]
Lista ordenada decrescente: [1, 2, 3, 4, 5]


## Iterando sobre uma Lista

In [ ]:
# Iterando sobre uma lista e imprimindo cada elemento
lista = ['a', 'b', 'c', 'd', 'e']
for elemento in lista:
    print(elemento)

In [ ]:
# Iterando sobre uma lista e imprimindo cada elemento
lista = ['a', 'b', 'c', 'd', 'e']
for elemento in lista:
    if elemento == 'a':
      print('Encontrei a letra...')

## Verificando a Existência de um Elemento

In [ ]:
# Verificando se um elemento está presente na lista
lista = ['a', 'b', 'a', 'd', 'e']
elemento = 'a'

if elemento in lista:
    print("O elemento", elemento, "está na lista.")
else:
    print("O elemento", elemento, "não está na lista.")

O elemento a está na lista.


## Retorna o número de elementos de uma lista

In [ ]:
fruits = ['orange', 'apple', 'pear', 'orange', 'kiwi', 'apple', 'banana']
fruits.count('orange')

2

In [ ]:
fruits = ['orange', 'apple', 'pear', 'orange', 'kiwi', 'apple', 'banana']
cf=0
for i in fruits:
  if i=='orange':
    cf+=1
print(cf)

2


In [ ]:
frutas = ['orange', 'apple', 'pear', 'orange', 'kiwi', 'apple', 'banana']
c=0
for i in frutas:
  if i[-1]=='e':
    c+=1

print('O total de palavras com inicio em -e- ',c)

O total de palavras com inicio em -e-  4


In [ ]:
lista = [5, 2, 4, 2, 2]
lista.count(2)

## Remove todos os elementos de uma lista

In [ ]:
lista = [5, 2, 4, 2, 2]
print(lista)
lista.clear()
print(lista)

In [ ]:
listap=[]
listai=[]

for i in range(210,512,2):
  listap.append(i)
  listai.append(i-1)

print('Lista dos valores pares:')
print(listap)
print('Lista dos valores impares:')
print(listai)

Lista dos valores pares:
[210, 212, 214, 216, 218, 220, 222, 224, 226, 228, 230, 232, 234, 236, 238, 240, 242, 244, 246, 248, 250, 252, 254, 256, 258, 260, 262, 264, 266, 268, 270, 272, 274, 276, 278, 280, 282, 284, 286, 288, 290, 292, 294, 296, 298, 300, 302, 304, 306, 308, 310, 312, 314, 316, 318, 320, 322, 324, 326, 328, 330, 332, 334, 336, 338, 340, 342, 344, 346, 348, 350, 352, 354, 356, 358, 360, 362, 364, 366, 368, 370, 372, 374, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 396, 398, 400, 402, 404, 406, 408, 410, 412, 414, 416, 418, 420, 422, 424, 426, 428, 430, 432, 434, 436, 438, 440, 442, 444, 446, 448, 450, 452, 454, 456, 458, 460, 462, 464, 466, 468, 470, 472, 474, 476, 478, 480, 482, 484, 486, 488, 490, 492, 494, 496, 498, 500, 502, 504, 506, 508, 510]
Lista dos valores impares:
[209, 211, 213, 215, 217, 219, 221, 223, 225, 227, 229, 231, 233, 235, 237, 239, 241, 243, 245, 247, 249, 251, 253, 255, 257, 259, 261, 263, 265, 267, 269, 271, 273, 275, 277, 279, 281, 283, 2

In [ ]:
listap=[]
listai=[]

for i in range(210,512,2):
  listap.append(i)
  listai.append(i-1)

print('Lista dos valores pares:')
print(listap)
print('A número de elementos..: ',len(listap))
print('A soma dos elementos...: ',sum(listap))
print('A média dos elementos..: ',sum(listap)/len(listap))
print('O maior elementos......: ',max(listap))
print('O menor elementos......: ',min(listap))

print('Lista dos valores impares:')
print(listai)
print('A número de elementos..: ',len(listai))
print('A soma dos elementos...: ',sum(listai))
print('A média dos elementos..: ',sum(listai)/len(listai))
print('O maior elementos......: ',max(listai))
print('O menor elementos......: ',min(listai))

## Questões desafios

### Questão 01

Um grande supermercado deseja implementar uma função no seu sistema de vendas que identifique e destaque os produtos que estão vendendo acima de uma certa quantidade mínima. Para isso, os programadores do supermercado têm uma lista com a quantidade de vendas diárias de cada produto. Eles precisam identificar quais produtos venderam acima de um limite estabelecido, e qual é o produto que teve o menor número de vendas entre os que excederam esse limite.


Você foi contratado para ajudar a implementar essa função.

Escreva um programa em Python que:

* Receba uma lista com as quantidades de vendas diárias de diferentes produtos (Cada elemento da lista representa a quantidade de vendas de um determinado produto).

* Filtre os produtos que tiveram vendas acima de um certo limite.

* Identifique e retorne o produto que teve o menor número de vendas entre os que excederam esse limite.

```
vendas = [120, 85, 300, 45, 210, 95]
limite = 100

# Sua tarefa é encontrar o produto com o menor número de vendas que está acima do limite.

```

**Saída Esperada**:

* Neste caso, o programa deve retornar 120, pois é o menor valor da lista entre os produtos que tiveram vendas acima de 100.

**Requisitos**:
* Utilize estruturas como loops, condicionais e listas.
* Deve ser considerado o caso de não haver nenhum produto que ultrapasse o limite (o programa deve indicar essa situação).

**Resposta Esperada**:

* O programa deve filtrar a lista e identificar o produto com menor venda, retornando-o. Caso nenhum produto atenda ao critério, deve exibir uma mensagem apropriada.

In [ ]:
# Seu código aqui.

120


### Questão 02

Uma empresa de tecnologia está analisando a performance dos desenvolvedores em um projeto, com base no número de linhas de código (LOC) que eles escreveram em um dia. A empresa deseja encontrar quais desenvolvedores escreveram linhas de código acima da média do time e, em seguida, identificar quem escreveu o maior número de linhas entre aqueles que estão acima da média.

Você foi chamado para ajudar a desenvolver essa funcionalidade.

Escreva um programa em Python que:

* Receba uma lista com o número de linhas de código que cada desenvolvedor escreveu em um determinado dia.
* Calcule a média de linhas de código escritas por todos os desenvolvedores.
* Filtre os desenvolvedores que escreveram acima da média.

Identifique e retorne o desenvolvedor que escreveu o maior número de linhas de código entre os que estão acima da média.


```
loc = [150, 200, 250, 180, 220, 175]
# A média de linhas de código escritas é de 195.83.
# Sua tarefa é encontrar o desenvolvedor com o maior número de linhas de código acima da média.

```
**Saída Esperada**:
* Neste caso, o programa deve retornar 250, pois é o maior número de linhas de código acima da média calculada.

**Requisitos**:

* Utilize estruturas como loops, condicionais e listas.
* Deve ser considerado o caso de todos os desenvolvedores estarem abaixo da média (o programa deve indicar essa situação).

**Resposta Esperada**:

* O programa deve calcular a média das linhas de código, filtrar os valores acima dessa média, e identificar o valor mais alto entre os que foram filtrados.

* Caso todos os valores sejam menores ou iguais à média, o programa deve exibir uma mensagem indicando que nenhum desenvolvedor superou a média.


In [ ]:
# Seu código aqui.